### PR TITLE
Alerting: Use new "Label" components for alert instance labels

### PR DESCRIPTION
**Pull request by gillesdemey**
**What is this feature?**

There has been a lot of confusion around "labels" and "label matchers".

Labels are simple `key + value` pairs and label matchers are `key + operator + value` objects that can "select" labels.

Technical differences aside, we weren't doing a great job in the UI to differentiate between the two. Showing both a label as `foo=bar` and an operator that selected all labels with key "foo" and value "bar" would also be shown as... `foo=bar`.

This PR adds support for `color` to the `Label` component used in Alerting which focuses its visual presentation on key-value pairs.

Additionally the colors are now determined by the "key" only, this makes it easier to spot what matcher will look at (and _might_ match depending on the operator) a particular label adding a visual indicator to the connection between labels and label matchers.

<details open="open"><summary>After</summary>
<p>
Screenshot 2023-06-30 at 17 52 44
</p>
</details> 

<details><summary>Before</summary>
<p>
Screenshot 2023-06-30 at 18 01 39
</p>
</details>

**notes for engineers / UX**

We have an opportunity now to also reconsider the list of colors to select from as this change will change the label colors anyway.

Should we use a different color scheme? Maybe one with more color values? What about a11y? Does the current list provide sufficient differences?

**Some reading material**
https://www.tableau.com/blog/examining-data-viz-rules-dont-use-red-green-together
https://davidmathlogic.com/colorblind/#%23000000-%23E69F00-%2356B4E9-%23009E73-%23F0E442-%230072B2-%23D55E00-%23CC79A7

### CODE REVIEW

**Comment by soniaAguilarPeiron** [comment]
> Should we use a different color scheme? Maybe one with more color values? What about a11y? Does the current list provide sufficient differences?
💯 Considering that approximately ten percent of men are colorblind, with red/green color vision issues being the most common, I believe it would be prudent for us to explore alternative color schemes to address this concern 👍

File: public/app/features/alerting/unified/components/AlertLabels.tsx [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
   labels: Record<string, string>;
-  className?: string;
+  size?: 'md' | 'sm';
 }
 
-export const AlertLabels = ({ labels, className }: Props) => {
-  const pairs = Object.entries(labels).filter(([key]) => !(key.startsWith('__') && key.endsWith('__')));
+export const AlertLabels = ({ labels, size }: Props) => {
+  const styles = useStyles2((theme) => getStyles(theme, size));
+  const pairs = chain(labels).toPairs().reject(isPrivateKey).value();
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/AlertLabels.tsx:17]: nice!

File: public/app/features/alerting/unified/components/Label.tsx [hunk 2]
```diff
@@ -10,52 +11,75 @@ interface Props {
...
     </div>
   );
 };
 
-const getStyles = (theme: GrafanaTheme2) => ({
-  meta: (color?: string) => ({
+const getStyles = (theme: GrafanaTheme2, color?: string, size?: string) => {
+  const backgroundColor = color ?? theme.colors.secondary.main;
+
+  const borderColor = theme.isDark
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/Label.tsx:38]: 🙌

File: public/app/features/alerting/unified/components/AlertLabels.tsx [hunk 3]
```diff
@@ -1,17 +1,41 @@
...
+}
+
+const isPrivateKey = ([key, _]: [string, string]) => key.startsWith('__') && key.endsWith('__');
+
+const getStyles = (theme: GrafanaTheme2, size?: string) => ({
+  wrapper: css`
+    display: flex;
+    flex-wrap: wrap;
+
+    gap: ${size === 'md' ? theme.spacing() : theme.spacing(0.5)};
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/AlertLabels.tsx:39]: what about lg size?

File: public/app/features/alerting/unified/components/AlertLabels.tsx [hunk 4]
```diff
@@ -1,17 +1,41 @@
...
   );
 };
+
+function getLabelColor(input: string): string {
+  return getTagColorsFromName(input).color;
+}
+
+const isPrivateKey = ([key, _]: [string, string]) => key.startsWith('__') && key.endsWith('__');
+
+const getStyles = (theme: GrafanaTheme2, size?: string) => ({
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/AlertLabels.tsx:34]: maybe in here we should define size as :` size?: 'md' | 'sm'`

File: public/app/features/alerting/unified/components/Label.tsx [hunk 5]
```diff
@@ -10,52 +11,75 @@ interface Props {
...
-        <div className={styles.meta().value}>{value}</div>
+        <div className={styles.value}>{value}</div>
       </Stack>
     </div>
   );
 };
 
-const getStyles = (theme: GrafanaTheme2) => ({
-  meta: (color?: string) => ({
+const getStyles = (theme: GrafanaTheme2, color?: string, size?: string) => {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/Label.tsx:35]: same in here ` size?: 'md' | 'sm'` ?

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM! 🎉
I added some non-blocking comments 👀

File: public/app/features/alerting/unified/components/AlertLabels.tsx (hunk 4 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/components/AlertLabels.tsx:34]: Good point, I should abstract this in to a separate union type

**Comment by gillesdemey** [comment]
I'm going to defer making any changes to the color palette for now as we simply use the same one as the existing `<Tags />` component and I'm hoping we can solve the accessibility concerns there.

**Pull request merged**